towhee/functional/mixins/column.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pyarrow as pa
+
+
+class ColumnMixin:
+    """
+    Mixins to support column-based storage.
+    """
+    def col_table(self):
+        """
+        Create a column-based table.
+
+        Examples:
+
+        >>> from towhee import Entity, DataFrame
+        >>> e = [Entity(a=a, b=b) for a,b in zip(['abc', 'def', 'ghi'], [1,2,3])]
+        >>> df = DataFrame(e)
+        >>> table = df.col_table()
+        >>> table
+        pyarrow.Table
+        a: string
+        b: int64
+        ----
+        a: [["abc","def","ghi"]]
+        b: [[1,2,3]]
+        """
+        header = [*self._iterable[0].__dict__]
+        cols = [[] for _ in header]
+
+        # cols = [[getattr(entity, col) for entity in self._iterable] for col in header]
+        def inner(entity):
+            nonlocal cols, table
+            for col, name in zip(cols, header):
+                col.append(getattr(entity, name))
+
+        any(map(inner, self._iterable))
```

Comment:
whats the point of using any() here?

Reply:
`inner()` does not return any value, so we need an `any()` or `all()` to execute `inner()`, otherwise `inner()` won't be executed.

Reply:
This is a dangerous action, if we accidentally change inner at some point, and introduce the possibility of returning true, the map wont be run completely. I think for this case using map is pointless, as we are generating useless data. Instead I would just use a regular for loop:
```python
for x in self._iterable:
   inner(x)
```

https://stackoverflow.com/questions/1080026/is-there-a-map-without-result-in-python/1080300#1080300
https://stackoverflow.com/questions/47722712/use-map-for-functions-that-does-not-return-a-value

Reply:
But I think we use `map()` here to realize stream computing. If we use for loop, we need to load the whole `self._iterable` which might cause OOM

Reply:
if the self._iterator is a list it will already be loaded into memory, if its an iterator then it will be iterated over (stream based) I believe.